### ISSUE TITLE
Ensure reactome go-cams validate against main GO shex schema

### ISSUE THREAD

**Issue by goodb** [geneontology/pathways2GO#69]
Revisit and rework conversion rules or shex schema to bring them into alignment.  Close when everything passes.  Re-open individual tickets for future failures.  

- [x] Implement re-usable software framework for testing models that is integrated with this repository's travis system
- [ ] Complete a clean validation run using this framework

**Comment by ukemi** [comment]
Fingers crossed!

**Comment by goodb** [comment]
Hit two issues so far.  

1. Need to ensure that protein complexes end up being subclasses of chemical entity.  I added an  assertion that molecular entity is a subclass of chemical entity to the Reactome Entity Ontology (RE).  Complexes are subclasses of molecular entity. With that assertion or a chebi import (that contains the same assertion) they thus get to be chemical entities as well so models that use them as inputs outputs etc. can validate.  This seems okay right now. 
2. The conversion uses 'causally upstream of' to link reactions/functions that are related by the 'nextEvent' relation in the biopax.  In some cases this is subsequently specified (e.g. to directly positively regulates) but in many cases it is not.  This breaks the shex model on the MF constraint:
causally_upstream_of: BiologicalProcess *;
as MFs aren't BPs..  I think this is an error in the shex schema - it looks like a current proposal would fix this: https://github.com/geneontology/go-shapes/issues/137 

Here is an example of the latter problem, which is very common.
![Screen Shot 2019-08-23 at 2 59 14 PM](https://user-images.githubusercontent.com/2086122/63626490-2d873200-c5b8-11e9-877f-f00b1ca90d5d.png)

**Comment by goodb** [comment]
@vanaukenk in relation to https://github.com/geneontology/go-shapes/pull/160https://github.com/geneontology/go-shapes/pull/160 I am running all of the current conversions through the validator.  It looks like about 25% of them do not pass as things stand today.  I will catalogue failures here as I diagnose them.  

Example #1 http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-8852276
There are cases in that one where a function is asserted to 'directly_provide_input_for' a process.  Note that the process in this case is not specified (this is the current resolution for transport) e.g. look for "GTSE1 promotes translocation of TP53 to the cytosol"

**Comment by goodb** [comment]
Another problem case.  http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-1482788  
It looks like at least one problem here is the constraint that functions be enabled by InformationBiomacromolecules (or complexes).  We have Set entities filling in the enabled_by slot that are only typed as 'chemical entities' (root).  Schema is fine here, need to make the typing for Sets better.

**Comment by goodb** [comment]
The conversion sometimes results in functions with more than one enabler, e.g. "PH receptors autophosphorylate" in http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-2682334  
The schema currently allows only 0 or 1 enablers.

**Comment by ukemi** [comment]
Interesting again! We are faithfully representing the info.

In the first case, the function is enabled by EPHRINAs or EPHRINBs. Biologically this is correct. Any set will do this, but the restirction is hidden because the set is a single entity.
Spot checking the annotations for some of the EPHRIN genes in AmiGO shows that current annotations support these individual gene products enabling the molecular functions.

In the second example, it is also correct. Either entity can enable the kinase activity. (I'm not going to go down the path as to whether this actually represents a process of phosphorylation that is made up of two molecular functions.  :)  )

However, I thought it was against Reactome curation policy to allow more than one active unit in a reaction. @deustp01? Would it have been more consistent to create a set in these cases: p-S172-TBK1 [plasma membrane] and p-S172-IKBKE [plasma membrane]?

I was actually hoping we could go the route of more than one active subunit to solve the BMP receptor complex issue. If we could enumerate all of the catalytic subunits of the complex when we ditch the contributes_to rule, we could get more info. This skirts the whole issue of 'emergent functions', but do our users really care about that? When they ask for a list of genes that can carry out a function, don't they want all the members of a catalytic complex? @vanaukenk   Complexes Complexes Complexes. Another alternative would be to propose that Reactome (and GO curators) somehow capture this info. Then how do we serve it up?

**Comment by goodb** [comment]
@ukemi @deustp01 it looks like recently introduced changes to the conversion have cured most of the problems above.  Out of a sample of 650 converted pathways, only 51 now fail the shex validation test.  (So about 92% success rate).  I checked 10 of the failures.  Of these 10, 3 failures were caused by having multiple enablers on a reaction node and untyped gene product nodes cause 7.  Examples of reactions with untyped gene product nodes enabling reactions include:
Asymmetric localization of PCP proteins   - unknown kinase
CS DS degradation.ttl  - "beta-xylosidase" R-HSA-2247521
Metabolism of ingested H2SeO4 and H2SeO3 into H2Se - R-HSA-5359054
Glyoxylate metabolism and glycine degradation - R-HSA-6784434
Signal transduction by L1  - Phosphorylation of VAV2
Aflatoxin activation and detoxification - Unknown NAT transfers COCH3 to AFXBO-C, AFNBO-C
Vitamin B1 (thiamin) metabolism - TDPK (looks like reaction has been changed in release 70 but still untyped)

When the converter encounters a node without any information beyond "physical entity", as in the cases above, it types it as a chebi 'chemical entity' (which is the root of all physical entities in our system).  The shex schema does not like functions to be enabled by anything aside from gene products or complexes.  

Thoughts on this problem?  I could infer that if an untyped node is enabling a reaction, it must be an information macromolecule and make that assignment when the models are created.  Would that be okay?

**Comment by ukemi** [comment]
Interesting, Peter? It might be a better strategy to type these in Reactome. Some look like black-box reactions.
1. Asymmetric localization of PCP proteins----Clearly the unknown kinase should be an enzyme/gene product or complex
2. CS DS degradation.ttl - "beta-xylosidase" R-HSA-2247521----is a  Genes and Transcripts [GenomeEncodedEntity] this should certainly be an information biomacromolecule.
3. Metabolism of ingested H2SeO4 and H2SeO3 into H2Se - R-HSA-5359054----is a  Genes and Transcripts [GenomeEncodedEntity] (PAPSe reductase) this should certainly be an information biomacromolecule.
4. Glyoxylate metabolism and glycine degradation - R-HSA-6784434----- This is a black-box reaction. Is it safe to assume it is an information biomacromolecule? peroxisomal glyoxylate carrier [peroxisomal membrane]
5. Signal transduction by L1 - Phosphorylation of VAV2-----Another black box unknown kinase. Safe to assume it is a biomacromolecule.
6. Aflatoxin activation and detoxification - Unknown NAT transfers COCH3 to AFXBO-C, AFNBO-C--------Another unknown NAT [cytosol] black box. Safe to assume it is a biomacromolecule?
7. Vitamin B1 (thiamin) metabolism - TDPK--------Looks like another enzyme where we know it exists, but don't know what it is. Should be a biomacromolecule.

**Comment by ukemi** [comment]
PS. would like Peter to comment too.

**Comment by ukemi** [comment]
If the ratio stays the same, it only looks like there are ~15 pathways with multiple enablers that will require new sets. I predict all the others are black-box reactions.

**Comment by deustp01** [comment]
Yes, David's predictions seem plausible. I will look carefully.

**Comment by deustp01** [comment]
> Out of a sample of 650 converted pathways, only 51 now fail the shex validation test. (So about 92% success rate)
@goodb could you send me the list of 51 so I can check?

**Comment by ukemi** [comment]
@goodb. Welcome to the world of the in-the-weeds curators!  :)

**Comment by goodb** [comment]
Examples in your emails.

**Comment by ukemi** [comment]
Thanks @goodb. @deustp01 do you want to look yourself, split them up or look together?

**Comment by deustp01** [comment]
I've done the first seven from the list that Ben added to this ticket earlier today.

Two conclusions. First, in all cases the problem is that we have asserted that a protein with no UniProt ID mediates a function defined by a proper GO molecular function term. Our goal is to associate functions with gene products, so this kind of annotation is rare - we should do it only when a reaction is known to happen even though the protein that mediates it has not been characterized enough to allow it to be identified, and we need the reaction to let us connect all the parts of a process together. We in fact have a separate class of physical entity, genomeEncodedEntity, used to annotate such unidentified proteins. So could this class distinction be made into a test to tell the ShEX QA that it's looking at an instance with missing information rather than an incorrect one?

EntityWithAccessionedSequence - the class that holds proteins that do have UniProt IDs - is a child of genomeEncodedEntity. We strictly enforce the rule that an EWAS must have a UniProt attribute, so any member of the GEE class that lacks a UniProt identifier must be a GEE.

Second conclusion / question. As you'll see from the details below, sometimes it looks like a catalystActivity instance is the target of the ShEX xomplaint. This is what I'd expect because this is the class that combines a specific entity with a specific molecular function so this is where unexpected kinds of physical entities should cause problems. But in some cases, reactions or whole pathways seem to have gotten flagged. This is not a big practical problem - as you've already noticed drilling down to the problem entity is easy. I'm just surprized by the irregulasr granularity.

Now the weeds.

1. Asymmetric localization of PCP proteins----Clearly the unknown kinase should be an enzyme/gene product or complex
Yes. In the same way that there is a reactionlike event class that has specific types of reactions as children (so GO-CAM could point to the parent when the exact reaction type is unknown or has no associated GO molecular function term, there is a genome-encoded entity class which has entities with accessioned sequences (EWAS) (i.e., canonical forms of proteins with UniProt IDs) as its major child class. However, it is also legal to create instances of the parent genome-encoded entity class when there is evidence that a protein (alone or as part of a complex) does something but not enough evidence to identify the protein uniquely, as here. How to detect this circumstance? All members of the EWAS class must have associated UniProt IDs – this is rigidly enforced – so any annotation of a macromolecule with no identifier must be an instance of the genome-encoded entity class.
Something I don’t understand about this instance is why the pathway is flagged while in problem cases 2 and 3 it’s the individual reaction. It looks like the ShEX issue is at the reaction level in all cases.

2. CS DS degradation.ttl - "beta-xylosidase" R-HSA-2247521----is a Genes and Transcripts [GenomeEncodedEntity] this should certainly be an information biomacromolecule.
Yes, exactly like the kinase in example 1.

3. Metabolism of ingested H2SeO4 and H2SeO3 into H2Se - R-HSA-5359054----is a Genes and Transcripts [GenomeEncodedEntity] (PAPSe reductase) this should certainly be an information biomacromolecule.
Yes, exactly like the kinase in example 1.

4. Glyoxylate metabolism and glycine degradation - R-HSA-6784434----- This is a black-box reaction. Is it safe to assume it is an information biomacromolecule? peroxisomal glyoxylate carrier [peroxisomal membrane]
Yes, exactly like the kinase in example 1. Another granularity oddity. In examples 2 and 3 it looks like the catalystActivity instance (the class that juxtaposes an entity and a molecular function to assert the entity enables the function) appears to be what was flagged by the ShEX script, and that’s right. In example 1, a whole pathway involving multiple binding reactions and one catalyzed reaction got flagged. Here (example 4) the whole reaction got flagged.

5. Signal transduction by L1 - Phosphorylation of VAV2-----Another black box unknown kinase. Safe to assume it is a biomacromolecule.
Yes, exactly like the kinase in example 1.

6. Aflatoxin activation and detoxification - Unknown NAT transfers COCH3 to AFXBO-C, AFNBO-C--------Another unknown NAT [cytosol] black box. Safe to assume it is a biomacromolecule?
Yes, exactly like the kinase in example 1.

7. Vitamin B1 (thiamin) metabolism - TDPK--------Looks like another enzyme where we know it exists, but don't know what it is. Should be a biomacromolecule.
Yes, exactly like the kinase in example 1.

**Comment by deustp01** [comment]
> Thanks @goodb. @deustp01 do you want to look yourself, split them up or look together?

I'll start by myself. If the pattern of the first seven holds up, we may already have all the information there is to get.

**Comment by goodb** [comment]
Apologies @deustp01 I must always remember to be more precise when sending spreadsheets to you..   The notes column is not a specific indication of what aspect of the model failed the shex, it was just a where I was jotting things down as I looked through them so that I had a trail to find my way back.  

All of the untyped gene product node cases are molecular function nodes failing against the rule that they can only be enabled by proteins or complexes.  Those failures have cascading effects on other nodes connected to the problem case by causal relations.  

When I look at the BioPAX for these pathways, I don't see any signs of the genomeEncodedEntity  class.  Unless there is an accession number, all I see in there is that it is a Physical Entity.

**Comment by deustp01** [comment]
I'll ask Guanming about how / whether kinds of macromolecule physical entities are distinguished in our BioPax export.  (No problem about the granularity

**Comment by goodb** [comment]
If you are looking at the models on noctua-dev we have a new toy for you.  When examining a model that you think may be invalid, turn on the reasoner and look for the red boxes.  Red boxes indicate nodes that are failing validation.  Clicking on the nodes will give you a geeky json representation of what went wrong.  (it will be blank if its the 2 enablers problem - known bug).  Try it on this one: http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-4608870 
The reaction in the middle is the culprit.  Notice how it also invalidates the reactions with causal relations leading into it.  

![Screen Shot 2019-09-30 at 1 52 33 PM](https://user-images.githubusercontent.com/2086122/65916251-dbd98080-e389-11e9-982e-4dccba6c1456.png)
![Screen Shot 2019-09-30 at 1 52 42 PM](https://user-images.githubusercontent.com/2086122/65916254-ded47100-e389-11e9-9ea3-900cca74a6a9.png)

**Comment by goodb** [comment]
@deustp01 here is what I believe is a complete list of pathways in Reactome release 69 that contain reactions that are enabled by more than one entity.  

[reactome_r69_multi_enabler_reactions.txt](https://github.com/geneontology/pathways2GO/files/3674470/reactome_r69_multi_enabler_reactions.txt)

**Comment by deustp01** [comment]
> I'll ask Guanming about how / whether kinds of macromolecule physical entities are distinguished in our BioPax export.

And the answer is, "For GEE, we hacked BioPAX a little bit by converting a GEE into a PhysicalEntity directly. This is the best approach I could find.The same is used for any Reactome PhysicalEntity instances that don’t have ReferenceEntities assigned.

So I guess for Reactome to GO-CAM conversion via BioPAX, these instances can be found by testing each incoming monomer physicalEntity to determine whether it has a UniProt or ChEBI reference. If not, then it is an empty placeholder, to be handled (? or not?) by the physical entity equivalent of the reactionLikeEvent placeholder for transformations that are out of scope for GO like dissociations.

**Comment by goodb** [comment]
If they have references, they are typed correctly in the BioPAX and the conversion, so that is handled.  

Its a shame BioPax lumps SmallMolecule in with the biological macromolecules as direct children of physical entity.  Otherwise we could equate physical entity with information biomacromolecule CHEBI:33695 which I think is the placeholder we are looking for.  

I suppose we could hack the biopax again with a comment for the genome encoded entities, but in this case it seems that it is probably not worthwhile to do so as the overall information gained is low.  

Could we use the information that these entities are catalyzing reactions to infer that they are biomacromolecules ?  Are we safe to assume that Reactome would not record an untyped small molecule catalyzing a reaction?

**Comment by ukemi** [comment]
It's a shame that we can't just equateor create a relationship between a GEE and a biomacromolecule. Would it be safe to do this in the BioPax itself?

**Comment by deustp01** [comment]
> Could we use the information that these entities are catalyzing reactions to infer that they are biomacromolecules ? Are we safe to assume that Reactome would not record an untyped small molecule catalyzing a reaction?

The answer is almost always, yes. Here is the one exception I know of: we have overloaded the concept of catalysis to assert that [a photon mediates the activation of rhodopsin](https://reactome.org/PathwayBrowser/#/R-HSA-2187338&SEL=R-HSA-74101&PATH=R-HSA-162582,R-HSA-372790,R-HSA-388396,R-HSA-418594). 

If abuses like this are really rare (I think they are), then they could be eliminated on the Reactome side by re-annotating this reaction and any other where we have assigned an enzyme role to a small molecule. Even if small molecules do catalyze reactions in the body (I think that's really rare), Reactome and GO both want to capture the functions of gene products so such a catalysis is out of scope for both. If either Reactome or GO needs the reaction for connectivity in a process, we can find another way to annotate it.

**Comment by goodb** [comment]
I don't see a way to do it without going beyond what is in the standard.  When we extended it to handle active units, we did so in comments..   Not really ideal.

**Comment by goodb** [comment]
@deustp01 this is really feeling more like a Reactome curation issue than a conversion issue.  In many cases there is clearly more knowledge about the mystery molecules than we would capture by 'information biomacromolecule' - e.g. we know its a kinase etc.  I wonder if it would be best to leave the conversion as it is, and push the task of adding references to higher level classes (protein, RNA, complex, small molecule) when exact references are unknown to the reactome curators?  e.g. if Photon was typed as SmallMolecule, even without a more specific chebi id, we would be all set.

**Comment by deustp01** [comment]
Face-to-face next week! (Guanming also points to our API as a way to retrieve the disambiguating information from us.)

**Comment by deustp01** [comment]
>  In many cases there is clearly more knowledge about the mystery molecules ...
I need to look at some cases, working from your spreadsheet. So far, what I see is teasers: we can infer that enzyme-mediated phosphorylation is going on but we can't identify the gene product with the enzyme activity, and I think that leaves us just short of being able to make an assertion  that is in scope for either Reactome or GO.

**Comment by ukemi** [comment]
We can make an assertion that there is an enzyme at work and that it is 'probably' catalyzed by an information biomacromolecule (protein or complex). Seems this is a strength in the sense that it is a definite query for something we know exists, but don't know what it is. It is highly unlikely that these reactions proceed spontaneously. Wouldn't you agree @deustp01

**Comment by deustp01** [comment]
@ukemi Yes, agreed. I was more going off on the tangent of information loss to conclude that absent a named gene product to associate with the molecular function, there is not much so a ShEX patch is good if it's safe and easy, but maybe not otherwise.

**Comment by deustp01** [comment]
> Examples in your emails.
I've gone through all the instances marked "fail" in the first column of Ben's table from 2 days ago. Here are the results as an Excel workbook. The first sheet is Ben's table with what I found in the comments column - mostly failures because we annotated a protein with no UniProt value. (These are cases where the available data make a good case that a protein enables a GO molecular function but don't allow the protein to be identified.) A few involved a similarly vague or generic small molecule, and in a few cases I couldn't find the problem eve with the help of the handy new analyzer feature that puts a red dot on problem things in the GO-CAM display.

Also one Reactome error: we asserted that SRC was the catalyst of a reaction but made a phospho-SRC entity the active unit responsible for catalysis. And a few more cases where a reaction has two enablers, all fixable by creating set instances to hold all the enablers or something else equally straightforward.

In the second sheet, I've taken all the failed rows and resorted them on the reason and notes columns to reinforce the point (already suspected by Ben and David) that there are recurring problems - unidentified kinase and ubiquitin ligase are popular, as is "photon" as a physical entity.

[reactome_shex_fails.xlsx](https://github.com/geneontology/pathways2GO/files/3683195/reactome_shex_fails.xlsx)

**Comment by goodb** [comment]
it looks like the last three errors are more of the same:

-  Insulin receptor recycling.ttl -> 2 untyped 'physical entity' enablers: IDA and protein tyrosine phosphatase
- Serotonin Neurotransmitter Release Cycle.ttl	 -> multiple enablers for the two SNARE binding reactions
- Interaction between L1 and Ankyrins.ttl	-> untyped physical entity enabler - 'unidentified protein tyrosine complex'

**Comment by ukemi** [comment]
This is great. The only tricky thing left is the photon.

**Comment by deustp01** [comment]
ChEBI now has a term for photon - 30312 - so it will become a simpleEntity (like glucose, ATP, etc.) in our next release.

**Comment by goodb** [comment]
Is there a resolution here @ukemi ?  We know what the errors are (and pretty cool that less than 8% of all the models had any error and that there appear to be only two kinds of errors).  But I'm not sure what the plan is for dealing with them.  For me, I'd like to see Reactome make use of an ontology such as CHEBI to annotate the untyped physical entities with upper-level ontology classes that exist outside of their internal schema.  If that happened and was reflected in the BioPAX we could handle that one easily in the converter.  

Likewise for the multi-entity enabler cases, I think that falls to them to work out whether or not they want to change their models.  (There are a total of 47 models like this).  

Positive news here is that for the use case of querying the system for the functions of specific genes, neither of these shex violations matters.  In the first, there are no genes and in the second, the queries ought to work fine.

**Comment by deustp01** [comment]
> For me, I'd like to see Reactome make use of an ontology such as CHEBI to annotate the untyped physical entities with upper-level ontology classes that exist outside of their internal schema.

Actually, nothing in our schema prevents using high-level ChEBI terms. In fact, in cases where we are intending to make a statement about something generic like trypsin enzyme cleaving a peptide bond involving a lysine residue regardless of the rest of the amino acid sequence, we already do that. (Trypsin enables the cleavage of a polypeptide into two peptides.)

As for photon, working through the list of small molecule failures will be a good way to go back and fix the cases where a generic is available (and if it makes chemical sense, to ask ChEBI for any missing generics).

**Comment by goodb** [comment]
Hi @ukemi and @deustp01 .  Wondering what the resolution here ought to be for the untyped physical entity problem.  In summary, we see some reactions that are catalyzed by molecules with no entity reference (e.g. no uniprot id).  These are typed simply as physical entities in the biopax, which could mean any of {Complex, DNA, DNARegion, Protein, RNA, RNARegion, Smallmolecule}.  The conversion process currently types these entities with  CHEBI:chemical entity - the root of all physical entities in GO ontology space.  The shex schema prescribes that any function/reaction only be enabled by / catalyzed by a @<InformationBiomacromolecule> OR @<ProteinContainingComplex>.  

So, to bring these conversions into alignment, we could:
1. assume that they are InformationBiomacromolecules if they catalyze a reaction and make that assertion during the conversion
2. change the shex schema to allow for the possibility of functions enabled by any physical entity
3. change the biopax export from reactome to attach higher level classifications (e.g. 'protein' CHEBI:36080 or 'information molecule' CHEBI:33695) via the entity reference field.  

For me, I like 3. if its possible.  
?

**Comment by deustp01** [comment]
Option 3 should work. On the spreadsheet reactome_shex_fails.xlsx from two weeks ago (above), all the instances that I labeled GEE can be re-classified as 'protein' CHEBI:36080, and there is a ChEBI instance for photon (now imported into Reactome and used to re-class our photon instances as simple entities (chemicals). One 'other entity - DNA' can be a 'information molecule' CHEBI:33695 and 12 other problems are artifacts of irregular annotation on our side, not problems with the physical entities themselves. That leaves less than five cases that can't be handled.

We could probably find a way on the Reactome side to get that GEE = CHEBI:36080 into the annotation so that Ben does not need another ad hoc patch for the GO-CAM import.

**Comment by deustp01** [comment]
Continuing in that direction, if we can get CHEBI:36080 protein associated with GEEs that are known to be proteins (but we don't know or can't specify which one), the same database hack and logic should let us associate CHEBI:16991 deoxyribonucleic acid and and CHEBI:33697 ribonucleic acid with GEEs that are known to be DNAs and RNAs respectively (but we don't know or can't specify which one). I expect we won't need to go as far as CHEBI:33695 information molecule - at that level of generality / ignorance we shouldn't annotate.

I expect we'd need to add these tags using the cross-reference attribute of the GEE class, so there's the weedy issue of whether this attribute gets into the BioPAX and can reliably be retrieved on the GO-CAM end.

The hack is that CHEBI instances are stored in the referenceMolecule class of Reactome, which is intended to hold the reference versions of small molecules (e.g., leucine, glycine) or precisely specified classes like amino acid, so maybe putting RNA, DNA, and protein terms in that class is overloading. Seems harmless but if this looks like a good way to go I'll check with Guanming.

**Comment by goodb** [comment]
From the external data consumer view, this would be a helpful increase in the information provided.  I can't speak to how it fits into the internal data structure, but hope that you can find a way.

**Comment by deustp01** [comment]
OK. It's on the to-do list. Result will either be a better-defined concern from Guanming or a set of cross-reference instances pointing Reactome GEE instances to CHEBI:36080 protein, in pathways for you to work with.

**Comment by goodb** [comment]
Hi @deustp01 just checking in here.  As I understand it, the ball is in your court for the shex validation issues that we know about right now.

**Comment by deustp01** [comment]
Yes. There was an attack of grant-writing. I'll have time the remainder of this week.

**Comment by deustp01** [comment]
OK. I’ve re-read the ticket, got Guanming’s approval for adding high-level ChEBI terms (protein, chemical entity, etc.) as cross-refs to the otherEntity and genomeEncodedEntity instances, and am getting to work on the actual tagging. otherEntities look tedious but straightforward, genomeEncodedEntities look easier once we figure out how to distinguish them from the referenceGeneProduct instances they are classed with. Indeed, if Ben has or can easily make a list of all problem genomeEncodedEntity instances that will be a useful shortcut (but not crucial – the search can be done here; it just needs developer-level search skills).

**Comment by goodb** [comment]
I don't think there is a way to distinguish the genomeEncodedEntities from the OtherEntitys in the BioPAX, but here is a list of all the untyped physical entities in the last release.  Hope that helps a little.  

[untyped_physical.txt](https://github.com/geneontology/pathways2GO/files/3825568/untyped_physical.txt)

**Comment by deustp01** [comment]
It does - thanks.

**Comment by deustp01** [comment]
All otherEntity (miscellaneous incompletely described molecular entities) instances now have crossReference attributes that point to something, generally something quite generic, in ChEBI.
GenomeEncodedEntity (informational macromolecular monomers that are not well-enough characterized to map to a unique UniProt instance or equivalent) instances remain to be done but with Ben's table from earlier today that should be do-able fairly quickly.

**Comment by deustp01** [comment]
Our plant / Gramene (rice pathway) colleagues, in annotating pathways, have created physicalEntity (otherEntity) instances like "cold temperature" and "long day photoperiod" (which turn up in untyped_physical.txt) together with conventional incompletely specified chemical entities,  and used them as inputs or regulators of reactions.

A proper cleanup is not likely to happen any time soon. 

@goodb Can I use ChEBI:50906 "role" (A role is particular behaviour which a material entity may exhibit) to create a cross-reference value for such otherEntity instances and will that safely trick the GO-CAM import process into thinking these are valid physical entities for shex purposes? I'm guessing / hoping that this will be OK - it's up to the Gramene people to decide whether this misleads their users and, if so, what to do.

**Comment by deustp01** [comment]
This is a follow-up to the discussion Wednesday (Path2GO call).

Clean-up in Reactome gk_central database should now be done. The problem instances were ones that could not be mapped 1:1 to a UniProt protein or Ensembl gene or RNA transcript or to a ChEBI chemical entity. OtherEntities is a catchall class of entities ("polypeptide", "amino acid") that can be classified to some extent using high-level ChEBI terms (e.g., ChEBI:36080 "protein"). All otherEntity instances in gk_central now have ChEBI identifiers as cross-references. 

GenomeEncodedEntities are informational macromolecules (or defined pieces of them created by processing of a transcription or translation product) that can't be mapped unambiguously to an instance in a reference database (e.g., a purified, well-studied enzyme whose amino acid sequence is unknown). Some of these instances are in Ben's list untyped_physical.txt; additional ones were retrieved with a Cypher query of our public database. All have now been supplied with cross-references to ChEBI terms for "protein", "DNA", "RNA", or somewhat more fine-grained ones ("tRNA") when appropriate.

Known limitations. First, as in the comment form earlier today, a small number of the otherEntity instances aren't really physical entities - I hope the added ChEBI terms in these cases will let them pass the shex requiremewnts without corrupting anything else. Second, I tried to be consistent in my use of high-level ChEBI terms but I'm sure there is some variability here. Nothing, I hope, is badly mislabelled and everything should now pass the shex requirements.

**Comment by goodb** [comment]
@deustp01 I am so sorry, I mistakenly generated that file with plant reactome biopax as the input.  Glad you figured out a better way, and really sorry if I wasted any of your time.  In case its still useful, here is the human untyped entity file.
[untyped_physical.txt](https://github.com/geneontology/pathways2GO/files/3849239/untyped_physical.txt)
 
I'm not sure if chebi:role is a good default for untyped entities.  We do already handle some cases of chebi entities that are roles and not entities - e.g. anti-inflammatory drugs https://www.ebi.ac.uk/chebi/chebiOntology.do?chebiId=CHEBI:35475 .  If you have to group things like "cold temperature" and "long day photoperiod" together I think it might be better to just leave them untyped.  For shex purposes, if something typed as a chebi:role is in a position where  we are expecting e.g. protein or information biomacromolecule, it won't validate anyway.

**Comment by deustp01** [comment]
No waste. The plant instances need to be fixed as well in the long run, and even in the short term it could be a neat demonstration of the generality of the Pat2GO scheme that it supports a reasonably clean, low-loss rice export as well as human.

I'll undo the ChEBI:role typings. Only a few plant instances are affected; all human and species-agnostic ones got valid chemical IDs, I think.

**Comment by deustp01** [comment]
@goodb The new untyped_physical.txt file includes set entities as well as individual entities. Even when all members of a set fall in the same chemical class, e.g., they are all proteins or small molecules, it is safer (and maybe logically required) to assign a cross-ref to the set itself to tell shex what kind of chemical it contains? I guess complexes escape this requirement because all our complexes have protein components so it is reasonable to type the complex as "protein" even if it also sometimes contains non-protein subunits? Or "complex" is its own type in shex?

**Comment by ukemi** [comment]
I thought GO protein-containing complex was also valid. Seems like this would fit better.

**Comment by ukemi** [comment]
I also don't think it is a bad thing to try this with the plant Reactome. Although I think it is out of scope for this project, it is a next logical step at extending this effort to other pathway representations. I would think that plant Reactome is a baby step that should prett much work out of the box with minimal tweaking. Let's create a roadmap in NYC for what steps we need to take once this first step is complete.

**Comment by deustp01** [comment]
> I thought GO protein-containing complex was also valid. Seems like this would fit better.

Do I need to add this term as an xref anyway to Reactome complex instances? Can't my reasoning above be used to tell shex that when it sees a Reactome complex and needs a classification, it should use  GO:0032991 "protein-containing complex", no manual case-by-case manual intervention needed?

Also, is it OK to create xrefs for some instances with ChEBI terms and for others with GO_cell_component terms? That won't break any consistency checks at GO-CAM? Except for the manual work involved, it's no problem on our end - these xrefs are just used to make links out to external resources from the web page that shows the details of the instance so a user browsing from instance to instance might wonder at the mixture of kinds of links but nothing should break.

**Comment by deustp01** [comment]
> I also don't think it is a bad thing to try this with the plant Reactome. Although I think it is out of scope for this project, it is a next logical step at extending this effort to other pathway representations. I would think that plant Reactome is a baby step that should prett much work out of the box with minimal tweaking. Let's create a roadmap in NYC for what steps we need to take once this first step is complete.

Clarification - I thought Ben has already done a Path2GO conversion for the rice annotations, and it came out mostly OK. If (and only if) that's true, then it's worth a sentence or two in the paper and in a grant proposal to point to it as evidence that the conversion process can be generalized beyond human. In the same paragraph with a sentence or two pointing out that as the conversion relies on BioPAX it can be generalized beyongh Reactome format to anything that can be converted to BioPAX.

**Comment by ukemi** [comment]
You are correct. I think at the Berkeley meeting @goodb did the initial conversion and the models were generated. However, I don't think they were validated. I think it would be worth mentioning, but I am trying to avoid scope-creep so that we can have a defined end-point to this first phase of the project.

**Comment by deustp01** [comment]
Agreed - no scope-creep; only what can be used as-is now (if anything).

**Comment by goodb** [comment]
@deustp01 there is already logic in the converter that will assign types to sets based on their components, so no need to do that manually from the conversion perspective.  

Regarding plant reactome, we got as far as successfully running the converter, without modification, on the plant reactome biopax export and generating go-cams for all the pathways.  Shex and OWL validation could be done trivially if we want, though should await any in-progress changes to address known problems.  Manual inspection of one pathway looked good.  Probably enough already for a sentence or two.  If we want to flesh the generality aspect out further, we could spend a bit of time running things on pathway commons again.

Here is ABA synthesis in rice from the conversion at the meeting.  
<img width="1185" alt="Screen Shot 2019-10-08 at 1 11 49 PM" src="https://user-images.githubusercontent.com/2086122/68961566-8b768080-0787-11ea-9eb9-9aa338ca2dfa.png">

**Comment by goodb** [comment]
> > I thought GO protein-containing complex was also valid. Seems like this would fit better.
> 
> Do I need to add this term as an xref anyway to Reactome complex instances? Can't my reasoning above be used to tell shex that when it sees a Reactome complex and needs a classification, it should use GO:0032991 "protein-containing complex", no manual case-by-case manual intervention needed?

Correct, no manual case-by-case is needed here.  Complexes are already detected and assigned to the protein-containing complex class.  I would suggest that any xrefs used for link outs here point to PRO or complex portal entities if they exist.  But I guess that is for the future.  

> 
> Also, is it OK to create xrefs for some instances with ChEBI terms and for others with GO_cell_component terms? That won't break any consistency checks at GO-CAM? Except for the manual work involved, it's no problem on our end - these xrefs are just used to make links out to external resources from the web page that shows the details of the instance so a user browsing from instance to instance might wonder at the mixture of kinds of links but nothing should break.

Yes, the OWL and shex logic should handle GO CC and CHEBI fine.  Use as you see fit.  The only tricky part here are the roles.  These can be used (e.g. the drug classes I mentioned earlier) but may require some intervention - I'll have to look to see what happens if they are added.

**Comment by deustp01** [comment]
roles - it's safe to leave this for the future

**Comment by deustp01** [comment]
Here is Ben's spreadsheet from November 14.

I've checked all items.

For genomeEncodedEntities (GEEs, informational macromolecules that cannot be associated with a UniProt or Ensembl ID) and otherEntities (just that - things known to be discrete physical entities but the information to map them to single reference chemicals or macromolecules is missing) I've either confirmed that a high-level ChEBI term (e.g., ChEBI:16991 "deoxyribonucleic acid" or even ChEBI:24431 "chemical entity) has been added, or for instances not found in previous searches of gk_central, I've added them.

Previously, I overlooked our polymer class. We use this mostly to annotate well-defined protein multimers like microtubules and collagen structures, but we also have polyguanosine. The presence of covalent bonds attaching the monomers to one another is optional. Almost all, I think can be tagged with ChEBI:36080 "protein" as a cross-reference. (For version 2, it might be interesting to ask the script to try to retrieve specific UniProt or ChEBI identifiers from the annotations of the polymer components and use them to classify the polymer more precisely, but that is definitely a refinement for later.)

I also previously overlooked a number of non-human GEEs, mostly viral and bacterial entities that serve to trigger processes of normal innate immunity. These should all have cross references now.

Per Ben's comment above, I have not added cross references to set instances but have indicated the appropriate high-level ChEBI term in the spreadsheet in case that's useful for checking validation results or for training the validator. (In some cases, e.g., a set of complexes of complexes, it's necessary to drill down several steps to find an identifier that establishes the chemical nature of the identifier.)
[untyped_physical_Hs.xlsx](https://github.com/geneontology/pathways2GO/files/3869851/untyped_physical_Hs.xlsx)

**Comment by goodb** [comment]
Looks good @deustp01 .  I think right now, we mainly need a response from Guanming about why these higher-level xrefs are not appearing in the curator-site biopax exports.  

Did you look through the reactions with multiple enabling entities ?  That was the other main class of problems with the shex.

**Comment by deustp01** [comment]
> Did you look through the reactions with multiple enabling entities ? That was the other main class of problems with the shex.

Oops, no. Point me to the list / comment and I will get to work on that.

**Comment by ukemi** [comment]
Based on the GPAD outputs, we may want to revisit any type of cardinality rules we have here. We will need @deustp01's expertise on how various classes of enzyme complexes work.

**Comment by goodb** [comment]
@deustp01 here is what I had found when last I looked. 

id	title
R-HSA-5696397	Gap-filling DNA repair synthesis and ligation in GG-NER
R-HSA-5685939	HDR through MMEJ (alt-NHEJ)
R-HSA-912446	Meiotic recombination
R-HSA-6782135	Dual incision in TC-NER
R-HSA-6783310	Fanconi Anemia Pathway
R-HSA-6782210	Gap-filling DNA repair synthesis and ligation in TC-NER
R-HSA-1445148	Translocation of SLC2A4 (GLUT4) to the plasma membrane
R-HSA-5655862	Translesion synthesis by POLK
R-HSA-5693565	Recruitment and ATM-mediated phosphorylation of repair and signaling proteins at DNA double strand breaks
R-HSA-5693607	Processing of DNA double-strand break ends
R-HSA-212300	PRC2 methylates histones and DNA
R-HSA-427413	NoRC negatively regulates rRNA expression
R-HSA-6807505	RNA polymerase II transcribes snRNA genes
R-HSA-1660499	Synthesis of PIPs at the plasma membrane
R-HSA-1660516	Synthesis of PIPs at the early endosome membrane
R-HSA-1660514	Synthesis of PIPs at the Golgi membrane
R-HSA-1660517	Synthesis of PIPs at the late endosome membrane
R-HSA-1362409	Mitochondrial iron-sulfur cluster biogenesis
R-HSA-1227986	Signaling by ERBB2
R-HSA-156842	Eukaryotic Translation Elongation
R-HSA-9033241	Peroxisomal protein import
R-HSA-6791226	Major pathway of rRNA processing in the nucleolus and cytosol
R-HSA-72203	Processing of Capped Intron-Containing Pre-mRNA
R-HSA-9027283	Erythropoietin activates STAT5
R-HSA-2559580	Oxidative Stress Induced Senescence
R-HSA-5213460	RIPK1-mediated regulated necrosis
R-HSA-416572	Sema4D induced cell migration and growth-cone collapse
R-HSA-212676	Dopamine Neurotransmitter Release Cycle
R-HSA-210500	Glutamate Neurotransmitter Release Cycle
R-HSA-181429	Serotonin Neurotransmitter Release Cycle
R-HSA-264642	Acetylcholine Neurotransmitter Release Cycle
R-HSA-8866652	Synthesis of active ubiquitin: roles of E1 and E2 enzymes
R-HSA-8866654	E3 ubiquitin ligases ubiquitinate target proteins
R-HSA-2682334	EPH-Ephrin signaling
R-HSA-2871796	FCERI mediated MAPK activation
R-HSA-450302	activated TAK1 mediates p38 MAPK activation
R-HSA-450341	Activation of the AP-1 family of transcription factors
R-HSA-450321	JNK (c-Jun kinases) phosphorylation and activation mediated by activated human TAK1
R-HSA-9013973	TICAM1-dependent activation of IRF3/IRF7
R-HSA-168927	TICAM1, RIP1-mediated IKK complex recruitment
R-HSA-937041	IKK complex recruitment mediated by RIP1
R-HSA-936964	Activation of IRF3/IRF7 mediated by TBK1/IKK epsilon
R-HSA-983695	Antigen activates B Cell Receptor (BCR) leading to generation of second messengers
R-HSA-1169091	Activation of NF-kappaB in B cells
R-HSA-1679131	Trafficking and processing of endosomal TLR
R-HSA-202433	Generation of second messenger molecules
R-HSA-909733	Interferon alpha/beta signaling

**Comment by deustp01** [comment]
Thanks - that's what I needed.

**Comment by deustp01** [comment]
I've done 21 pathways on Ben's list. There appear to be only a few kinds of problems, which recur regularly as itemized in the attached Word document (with color coding that gets lost in a copy-paste to this comment window).
From what I know of our curation styles / variability this sample should be enough to reveal most of the ways that we've found to break parsing of a Reactome catalystActivity attribute into GO-CAM so I'd like to turn to something else in preparation for the meeting.
@goodb  But let me know if it's important to get through the entire list now.
[events_with_mutiple_enablers.docx](https://github.com/geneontology/pathways2GO/files/3880371/events_with_mutiple_enablers.docx)

**Comment by goodb** [comment]
@deustp01 I don't think you need to go through the whole list right now.  I had hoped we might make it to a clean release before the meeting, but that doesn't seem feasible at this point.  For now, I think its more important to work on defining the general patterns where things don't quite line up as it stands than to fix every instance.  

I was a little unclear about two of your error categories though.  Right now, the shex rules allow for the possibility of a Complex in the enabled_by slot for a reaction.  So the grey (catalyst is a homomultimer) and yellow (catalyst is a complex) tags shouldn't indicate a shex-specific problem.  They are probably most relevant for the GPAD generation, right??

**Comment by deustp01** [comment]
> They are probably most relevant for the GPAD generation, right??

I don't know. I was cataloguing all the places where the active unit of the physical entity that enables catalysis molecular function is anything other than a UniProt protein and, indeed, going through the list there are no pathways all of whose issues are gray or yellow - they all have some green or purple as well, so it looks like our focus is only the green (sets as active units) and purple (more than one catalyst or active unit per reaction) ones. That makes sense, and points the gray and yellow instances to "Systematic Review of Pathway Conversion"  ##40.

**Comment by goodb** [comment]
Okay, on the same page.  I was thinking about GPAD generation as these kinds of structures currently would not yield any gpad-style annotations to the proteins in the complexes and perhaps they should.

**Comment by ukemi** [comment]
I think you are correct, but this is happening in other scenarios as well. We should be sure to touch on this in the complexes discussion.

**Comment by goodb** [comment]
Dec. 18 shex/owl validation test
Using new Reactome model collection, released late november.  Using current converter - noting the return of Transport processes.  Using pending shex file here: https://github.com/geneontology/go-shapes/pull/172 
1806 models tested
197 have a shex validation error
0 have an OWL error

Many errors are caused by regulates relationships linking MFs to BPs 

I think most other errors come from untyped entities including:
inputs:
R-ALL-9634466 D5v8 (herceptin)
R-ALL-9665971 pertuzamab

enablers:
R-HSA-175986 Thiol S-methyltransferase
R-HSA-9661701 UBGNO 
R-CPE-9661737 UBGNR
R-CPE-9661715 BILR 
R-HSA-9663470 BGET
R-ALL-72617 eIF1

Looking at the untyped entities coming from the Reactome November 2019 build, there are quite a few and it seems that at least some that Peter has fixed did not make it through into this release.  Not sure if that is a timing issue or a technical one.  example: R-ALL-157771 ought to have the type 'protein' but that didn't come through in the latest release.  Here is the current untyped list.  

[untyped_physical.txt](https://github.com/geneontology/pathways2GO/files/3985103/untyped_physical.txt)

**Comment by ukemi** [comment]
The first two are monoclonal antibody cancer therapeutics. Are they inputs in normal patheways @deustp01? They would be proteins.

The others all look liked they can be typed too.

**Comment by ukemi** [comment]
Look on the bright side------no logical errors.

**Comment by ukemi** [comment]
I see now, the 2 mabs are annotated as inhibiting the receptor functions in the signaling pathways. When I look in the pathway browser, they appear to be in a nonphysiological patway, but if I just search on them it looks like they are part of the pathway.

**Comment by deustp01** [comment]
Hmm. The theory is that drugs are used to treat diseases, so drug instances (small molecule or protein or engineered protein) should not be appearing in normal events, and the Path2GO process should not be attempting to process any disease events. And even if a "normal" event involves a drug, it's not truly "normal" in the sense that all of normal reaction space exists and operates without any drugs as participants.

I will look some more at how we've annotated these to see if there's a pattern that would enable Path2GO to recognize and exclude these pseudo-normal reactions.

**Comment by deustp01** [comment]
> I think most other errors come from untyped entities including:
> inputs:
> R-ALL-9634466 D5v8 (herceptin)
> R-ALL-9665971 pertuzamab

@goodb These two instances point to a Reactome data structure oddity. It is permissible to include a "disease" reaction in a "normal" pathway. We do this to annotate actions of drugs that interact with normal proteins performing their normal functions, e.g., small molecules that inhibit some feature of blood clotting. You already use our disease attribute at the pathway level to identify pathways with non-null slots and not process them. An additional step (easy to say) would be to apply that test again to identify and remove "disease" reactions within normal pathways. A test that probably overlaps with the first one is to discard any reaction whose participating entities include ones that belong to the chemicalDrug or proteinDrug classes. (In principle, the two tests should flag the identical reactions - in reality??)

I expect that removal of all these disease reactions would leave a normal reaction set that still has the full connectivity of the pathway - yell if that's not true! Even if we've avoided that trap, this way of annotating drug functions may not be maintainable so anything that comes out of the Path2GO project that suggests additional problems it causes would be good to know to help us figure out a better annotation strategy.

**Comment by ukemi** [comment]
Maybe easier said than done. We can't use the ChEBI role drug or we will remove all reactions that use ATP.  :) Do you know of another way to identify drugs more rigorously?

**Comment by deustp01** [comment]
I'm suggesting we use the Reactoime annotation data themselves: our identification of a reaction as a disease event even though it is part_of a normal pathway, and / or the presence of a physical entity of the chemicalDrug, proteinDrug, or RNADrug Reactome classes as a participant in the reaction.

Agreed, looking up Reactome entities in ChEBI to identify their ChEBI-assigned roles would not work.

**Comment by ukemi** [comment]
Sweet. The power of annotation.

**Comment by goodb** [comment]
@deustp01 Actually I don't use the Reactome disease annotation slot for pathway filtering because it does not come through in the BioPAX.  I have hackily filtered out the disease pathways simply by looking for occurrences of 'disease' in the pathway name.  This annotation also does not come through at the Reaction level.  So.. if we wanted to make use of it in the converter, we would need to talk to Guanming about encoding it somehow.  It could perhaps be added as an additional xref linking to some appropriate ontology class or (less good) another parsable comment.  

If we do attempt to filter at the reaction level, it looks like this will be disruptive to some pathways.  As an example, I'm looking at reaction R-HSA-1963589 ERBB2 forms heterodimers with ligand-activated ERBB receptors.  It has a reference to pertuzamab.  If we took that reaction out of the Signaling by ERBB2 pathway, we would lose connectivity from its upstream 'ERBB3 binds neuregulins ' to its downstream 'SRC family kinases phosphorylate ERBB2'

![Screen Shot 2019-12-20 at 9 28 20 AM](https://user-images.githubusercontent.com/2086122/71277980-2c1e0a00-230b-11ea-8b0b-f751aef60362.png)

**Comment by ukemi** [comment]
Rats. I agree we can't remove that reaction. Would it work to remove anything with 'disease' or 'drug'? Still hacky but might work.

**Comment by goodb** [comment]
As far as the code for the converter is concerned, I don't see why it should not be able to handle pathways involving drugs and diseases.  We might or might not want to avoid including them in the GO knowledge base but that is a separate issue.  

I think we should focus first on getting these entities typed.  I think the IUPHAR annotations ought to make that possible one way or another.

**Comment by deustp01** [comment]
> As an example, I'm looking at reaction R-HSA-1963589 ERBB2 forms heterodimers with ligand-activated ERBB receptors. It has a reference to pertuzamab. If we took that reaction out of the Signaling by ERBB2 pathway, we would lose connectivity from its upstream 'ERBB3 binds neuregulins ' to its downstream 'SRC family kinases phosphorylate ERBB2'

True, but that regulatory interaction is specifically due to the drug interaction - no drug, the normal target reaction proceeds unregulated, so I think it's OK to lose it as a piece of "normal" biology.

**Comment by ukemi** [comment]
But even though the reaction is negatively regulated by the drug, the R-HSA-1963589 reaction itself represents the normal receptor-ligand interaction doesn't it?

**Comment by deustp01** [comment]
> I think we should focus first on getting these entities typed. I think the IUPHAR annotations ought to make that possible one way or another.

All our chemicalDrug instances have IUPHAR instances as referenceEntity attributes, as well as ChEBI instances as cross-refs, and presence of both attributes is an editorial requirement. All eight of our proteinDug instances (all monoclonal antibodies) have IUPHAR referenceEntity attributes. Are goal (achieved so far) is to have an IUPHAR reference for every drug we release. Is that information accessible in the BioPAX export, and usable here to identify reactions that are unwanted for now?

**Comment by ukemi** [comment]
I think that the problem is that since it references the pertuzamab as a negative regulator, a simple filtering of any reaction mentioning pertuzamab won't work.

**Comment by ukemi** [comment]
I knew this was going to be harder than it looked.   :)

**Comment by ukemi** [comment]
If the IUPHAR info is there, that should do the trick. A Reactome curator would not have assigned and IUPHAR value unless the entity was acting as a drug, right?

**Comment by deustp01** [comment]
> But even though the reaction is negatively regulated by the drug, the R-HSA-1963589 reaction itself represents the normal receptor-ligand interaction doesn't it?

No. Normally, ERBB2:ERBIN:HSP90:CDC37 interacts with Ligand-Activated EGFR/ERBB3/ERBB4 to form an array out outputs. Alternatively ERBB2:ERBIN:HSP90:CDC37 can bind any of an array of drugs to form complexes that have no further function, sequestering ERBB2:ERBIN:HSP90:CDC37. In the specific case where ERBB2:ERBIN:HSP90:CDC37 binds pertuzumab drug, the resulting complex inhibits the normal binding reaction, so it not only sequesters a required input to the normal reaction but also prevents other unsequestered copies of the input complex from doing their normal binding.

**Comment by deustp01** [comment]
> If the IUPHAR info is there, that should do the trick. A Reactome curator would not have assigned and IUPHAR value unless the entity was acting as a drug, right?

Yes. It's not perfect because it's an editorial standard not a hard-wired feature of the data model but we are working to keep this usage clean exactly as you say.

**Comment by ukemi** [comment]
> No. Normally, ERBB2:ERBIN:HSP90:CDC37 interacts with Ligand-Activated EGFR/ERBB3/ERBB4 to form an array of outputs. Alternatively ERBB2:ERBIN:HSP90:CDC37 can bind any of an array of drugs to form complexes that have no further function, sequestering ERBB2:ERBIN:HSP90:CDC37. In the specific case where ERBB2:ERBIN:HSP90:CDC37 binds pertuzumab drug, the resulting complex inhibits the normal binding reaction, so it not only sequesters a required input to the normal reaction but also prevents other unsequestered copies of the input complex from doing their normal binding.

Got it!

**Comment by deustp01** [comment]
> enablers:
> R-HSA-175986 Thiol S-methyltransferase
> R-HSA-9661701 UBGNO
> R-CPE-9661737 UBGNR
> R-CPE-9661715 BILR
> R-HSA-9663470 BGET
> R-ALL-72617 eIF1
> 
> Looking at the untyped entities coming from the Reactome November 2019 build, there are quite a few and it seems that at least some that Peter has fixed did not make it through into this release.

The first item in the list of enablers has a ChEBI xref in Reactome; the rest do not, probably because I overlooked them in the re-edit last month to add high-level ChEBI terms as xrefs to Reactome physicalEntities that are incompletely specified, e.g., we know it's a protein but not enough about its sequence to map it to a UniProt entry.

All the other items do have ChEBI xrefs, e.g., to ChEBI:36080 "protein". That's also true for the first 20 non-set items listed in untyped_physical.txt, so these cross-refs are not getting through to this release. (I only checked these 20 but they look representative.)

**Comment by goodb** [comment]
@deustp01 I just revisited the thread with Guanming and asked him how we might access this information.  I can see it there in the curator site, but not in the public one or in any of the BioPAX.  I suspect that he will need to make a specific release for us.  

Let me know what you two think about including/excluding reactions containing drugs.  If we want to go that route, I will set up the IUPHAR-based filter.

**Comment by deustp01** [comment]
@ukemi For Path2GO version 1, I think we exclude reactions that have drugs as participants, but bringing them in at least on a pilot scale, maybe as a first step towards bringing in disease processes / abnormal phenotypes, is a high priority for version 2 (and a discussion point for version 1, that the Path2GO tool can easily accommodate this extension).

**Comment by ukemi** [comment]
I agree completely with @deustp01. Including the reactions with drugs goes against general GO policy and is outside of scope. In the long term, this might be a very exciting extension of the project that goes beyond GO but moves further along the path towards complete integration of multiple resources.

**Comment by goodb** [comment]
Update.  (Ping @vanaukenk regarding shex issues below).

I ran the conversion using the current rule set (which includes the use of transport and protein complex disassembly as types for certain reactions), against a January 2020 version of Reactome generated from the curator (pre-release) database.  

1806 models processed . Zero OWL errors.  517 with some shex problem.  

Here is a [visualization of the shex conflicts](https://docs.google.com/presentation/d/1FW-j4q-LwDKVT_tk688_bDwtl7dT1ASCv-rTg-xqtK0/edit#slide=id.p):

Key Problems:

1. In the conversion, complexes are used as inputs, outputs, and objects of transport for Biological Process nodes.  (Either transport or dissociation processes).  This conflicts with the schema for Biological Process which only allows chemical entities or anatomical entities as inputs/outputs and only allows Information Biomacromolecules (genes, proteins) to be transported.
2. In the conversion, Molecular Function nodes are assigned causal (regulates, provides input for) relations to Biological Process nodes.  This violates the MF schema which dictates that these relations only connect to other MF nodes.  
3. We still have a variety of untyped physical entities - many of which seem to be Sets of mixed type.  

I am going to work on 3. between now and the meeting tomorrow (as we will likely mostly talk about physical entities).  @vanaukenk can you confirm whether or not the shex schema might change with regards to 1.?  I believe our resolution on the shex call last week was that its going to stay as it is for the MF-BP relations (2. above) - assuming that is still the case, @ukemi could you check the model proposed here: https://github.com/geneontology/pathways2GO/issues/75#issuecomment-572252975 ?

**Comment by vanaukenk** [comment]
@goodb 
Wrt 1, it seems reasonable to me to allow complexes to be inputs, outputs, and objects of transport for Biological Process nodes.
For 2, that is my understanding as well.  We should discuss the proposed new model in #75 on the 2020-01-15 call.

**Comment by deustp01** [comment]
In Ben's list of key problems, I don't understand the first one -

> In the conversion, complexes are used as inputs, outputs, and objects of transport for Biological Process nodes. 

@goodb could you point to an example to help me see what the problem is?
The second one -
> In the conversion, Molecular Function nodes are assigned causal (regulates, provides input for) relations to Biological Process nodes.
... could be arising because it is legal to manually assert that any event (reaction / function or pathway / process) precedes any other. We really shouldn't do this so, to the extent that that is the problem here, a clean-up is the way to go. But I don't see how to create an assertion that a reaction / function regulates a process so an example to help see where the problem is coming from would be useful.

**Comment by goodb** [comment]
For 1.  @deustp01 here is an example from the Scavenging by Class H Receptors pathway.  (Note that what you see here may be different and is a more recent version than what you see on noctua-dev.)

![Screen Shot 2020-01-14 at 3 13 11 PM](https://user-images.githubusercontent.com/2086122/72390819-8d55b500-36e0-11ea-8f62-9d8220a43808.png)

Here, the reaction "STAB2:ligand is endocytosed" is typed by the converter as a Transport biological process.  This does not cause any problems for the OWL reasoner, but introduces a number of violations of the schema that is being developed to define the overall rules for the structure of GO-CAM models.  (Hence the red dot).  The schema currently prevents complexes from being used as the objects of relations like has input, has output, and transports or maintains localization of when the subject is a biological process such as a transport.  

For 2.  The same pathway provides another example where a binding reaction has a causal relation to transport process.  As things stand, we have a number of rules in the converter that infer regulatory relationships (beyond 'causally upstream of' which we get from the nextStep relations in the BioPAX).  These generate the other kinds of relationships between functions and these process nodes.  
![Screen Shot 2020-01-14 at 3 19 16 PM](https://user-images.githubusercontent.com/2086122/72391306-a57a0400-36e1-11ea-9ae8-3b723b8c7628.png)

**Comment by ukemi** [comment]
I think an accurate representation of biology requires that complexes be allowed as transport objects. PMID:9852151, PMID:10545491, PMID:11102514.

**Comment by deustp01** [comment]
> Here, the reaction "STAB2:ligand is endocytosed" is typed by the converter as a Transport biological process.

Transport is correct but why is this Reactome event getting typed as a transport process rather than as a reaction mediated by a transporter function? Probably this is happening because the Reactome reaction ["STAB2:ligand is endocytosed"](https://reactome.org/PathwayBrowser/#/R-HSA-2173782&SEL=R-HSA-2247511&PATH=R-HSA-5653656) is missing information. We could have assigned a molecular function like  [GO:0005044 scavenger receptor activity](https://www.ebi.ac.uk/QuickGO/term/GO:0005044) by overloading our catalystActivity class to allow GO cargo receptor activity terms (just as we could do to explicitly annotate binding reactions).
Short-term fix - maybe none is needed (we can tolerate schema violations temporarily?)
Long-term fix - this is part of a larger project that both Reactome and GO need to develop a reaction typology (binding vs catalysis vs transport vs etc.) and implement it for Reactome curation.

**Comment by ukemi** [comment]
I @goodb. I have added a third model to #75 for discussion.

**Comment by goodb** [comment]
Update.  Following recent changes for #75 and #84 , I tried to run all the models through the shex/owl validation over the weekend.  

On the down side, things have really slowed down.  Only 1122 models completed the validation over the span of about 40 hours.  It seems the expansion of the shex schema is hitting performance pretty harder (to do list item to try to improve that).  

On the up side, we are getting close.  Of those 1122 models, all passed the OWL test, and only 35 failed the shex validation.  

Model R-HSA-189483 failed to validate because it uses an unknown GEE https://reactome.org/content/detail/R-CPE-9661737 - apparently from Clostridium.  

Everything else that I looked at failed because there is a reaction with more than one enabling molecule.  

Here is a list of all of the reactions with the multiple enabler problem (based on curator-version BioPAX from Jan 8, 2020):
[Multiple Enablers.txt](https://github.com/geneontology/pathways2GO/files/4118633/Multiple.Enablers.txt)

**Comment by ukemi** [comment]
> Model R-HSA-189483 failed to validate because it uses an unknown GEE https://reactome.org/content/detail/R-CPE-9661737 - apparently from Clostridium.

Fascinating! I need to think about this one.

**Comment by goodb** [comment]
For shex, it would validate with the fix we've used elsewhere of xrefing it to chebi:protein.  How the multi-species part of this goes, not sure.

**Comment by ukemi** [comment]
Is it the first step towards modeling microbiome interactions? Might be interesting for @cmungall and @vanaukenk. I think your shex solution is reasonable, it's the scope that I am pondering.

**Comment by deustp01** [comment]
Hmm. What's going on here is that bilirubin glucuronides transported from hepatocytes into the bile, and into the gut lumen, [are acted on sequentially by three bacterial enzymes](https://reactome.org/PathwayBrowser/#/R-HSA-189445&SEL=R-HSA-9661745&PATH=R-HSA-1430728), beta-glucuronidase (GUSB), biliverdin reductase (BILR), and UBGNR to form stercobilin.

<img width="855" alt="Screen Shot 2020-01-27 at 3 16 56 PM" src="https://user-images.githubusercontent.com/12737623/73210367-212f7400-4118-11ea-8ef7-f88b341a0771.png">

We annotated C. perfringens GUSB (UniProt Q8VNV4 - a TrEMBL instance) as the enzymes mediating the first reaction. We couldn't identify specific gene products to assign as enzymes to mediate the second and third reactions so instead created instances of the genomeEncodedEntity class to annotate them, all on the strength of evidence from [PMID:16504607](https://www.ncbi.nlm.nih.gov/pubmed/16504607).

Looking back, the evidence that these reactions happen in the gut is solid but I don't know how solid the evidence is that [C. perfringens](https://en.wikipedia.org/wiki/Clostridium_perfringens), better known as the causative agent of gas gangrene and a major causative agent in human food poisoning) is the main source of the enzymes that mediate them in healthy humans.

So there may be an annotation issue here. It's possible (to be discussed with curators) that we should retreat to asserting that unknown bacterial proteins catalyze these reactions, with no attempt to identify a bacterial species. Ben's suggestion of a chebi:protein xref looks like a good way out of the immediate shex issue (with an acceptably low amount of information loss, I think).

On the future of microbiome annotation, the odd annotation here points to a bigger problem, that genomic diversity within a bacterial species can be very large, so associating specific molecular functions with specific proteins will often be hard and even when it can be done, the result will often be a protein set (in the Reactome sense of set). Does this also sort of say something about scope?

**Comment by ukemi** [comment]
It says something very big about scope and how scope may differ. Is the GO scope of human annotation to annotate the function of human gene products, or is it to annotate any gene product used in a normal human physiological process? Same for Reactome....

**Comment by ukemi** [comment]
Note that annotating C. perfringes gene products is not out of scope for GO. Example:
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:Q8XLE8

**Comment by deustp01** [comment]
> Same for Reactome...

Reactome is inclusive: if it affects human biology, we're prepared to annotate it with the proviso (shout-out to Judy here) that we don't get credit against our annotation metrics for non-human proteins

**Comment by deustp01** [comment]
> Note that annotating C. perfringens gene products is not out of scope

Sure - handling symbiosis is the complicated bit, especially near the normal physiology : disease process boundary.

**Comment by goodb** [comment]
One thing @deustp01 's comment suggests to me is that it its going to be increasingly important to think about annotating at the gene family level rather than the individual gene level when that is the level of evidence.  FWIW..

Here are the only entities that are currently completely unclassified and thus schema breakers:
R-HBV-8982481 viral double-stranded RNA
R-HCV-8982462	viral double-stranded RNA
R-HSA-5357857	p-T357,S358-MLKL oligomer
R-MMU-9670591	Igkc
R-HSA-446069	Keratin 5/14
R-HSA-9660029	Poly-vimentin
R-HSA-9660015	Poly-vimentin
R-HSA-9657899	Poly-vimentin
R-HIV-173644	Multimeric capsid coat
R-HIV-175338	Multimeric matrix layer
R-HSA-2514779	Cleaved fibrillin-2
R-CPE-9661737	UBGNR
R-CPE-9661715	BILR
R-HSA-9661701	UBGNO
R-HSA-9663470	BGET
R-ALL-9638756	RNA(n+1)

**Comment by deustp01** [comment]
> One thing @deustp01 's comment suggests to me is that it its going to be increasingly important to think about annotating at the gene family level rather than the individual gene level when that is the level of evidence. FWIW..

Immediate practical patch question - I haven't looked up any of these instances yet to see what's going on - some look like things I should have fixed in the xref cleanup late in the fall. Would adding a ChEBI:protein (or ChEBI:RNA) xref to each of these fix the schema-breaking problem? 

Longer-term (version 2) technical issue: Reactome has a class "polymer" that up to now we have been able to evade in exports to GO-CAM. Poly-vimentin is an example. For version 1, I can xref it to ChEBI:protein, and we can discuss later how to handle a polymer instance whose monomers are a UniProt protein.

Ben's point about classes. As an editorial issue, we don't handle classes of macromolecules (e.g., "collagens", or "kinases") well because our goal is to annotate individual gene products. Our work-around to date has been the creation of sets, with defined and candidate set types as a way of handling groups with more or less evidence to support the grouping. The PRO ontology structure provides a way to identify protein families and to associate specific proteins with those families that may be a better approach to this problem. A version 2 discussion item, I think.

**Comment by goodb** [comment]
Yes, adding the CHEBI xrefs would solve the schema problem right now.

**Comment by deustp01** [comment]
> Here are the only entities that are currently completely unclassified and thus schema breakers:

All now fixed with ChEBI cross-references:
R-HBV-8982481 viral double-stranded RNA	67208 dsRNA
R-HCV-8982462 viral double-stranded RNA	67208 dsRNA
R-HSA-5357857 p-T357,S358-MLKL oligomer	36080 protein
R-MMU-9670591 Igkc	curation mistake
R-HSA-446069 Keratin 5/14	36080 protein
R-HSA-9660029 Poly-vimentin	36080 protein
R-HSA-9660015 Poly-vimentin	36080 protein
R-HSA-9657899 Poly-vimentin	36080 protein
R-HIV-9657899 Multimeric capsid coat	wrong ID in GO-CAM?
R-HIV-173644 Multimeric capsid coat	36080 protein
R-HIV-175338 Multimeric matrix layer	36080 protein
R-HSA-2514779 Cleaved fibrillin-2	36080 protein
R-CPE-9661737 UBGNR	36080 protein
R-CPE-9661715 BILR	36080 protein
R-HSA-9661701 UBGNO	36080 protein
R-HSA-9663470 BGET	36080 protein
R-ALL-9638756 RNA(n+1)	83400 RNA polyanion

**Comment by goodb** [comment]
Wish I could rerun now, but will either have to wait for this to percolate through the release or ask for another custom export from the curator site (which I hesitate to do).

**Comment by deustp01** [comment]
Let me see whether I can make any progress on the list of multiple enablers before we decide what to do here.

**Comment by deustp01** [comment]
@goodb Here's an interesting multiple-enablers case:
Activated JNKs   phosphorylate c-JUN [ R-HSA-168136](https://reactome.org/content/detail/R-HSA-168136)

We're asserting here that each of the three members of a set of proteins is capable by itself of enabling the molecular transformation annotated in the reaction. To do that, we've made the set the physical entity attribute of the reaction's catalystActivity, and made each of the set members an activeUnit attribute of that catalystActivity. If we can do this annotation consistently, could it be parsed correctly at the GO-CAM end, i.e., if the physical entity associated with catalysis is a set and the active units associated with that catalysis are all the members of the set, can this be interpreted to mean that set member 1 enables the annotated GO molecular function, set member 1 enables the annotated GO molecular function, ..., set member n enables the annotated GO molecular function?

R-HSA-168162 is another example of this.

**Comment by goodb** [comment]
@deustp01 erm..   This seems like an inappropriate use of the active-unit slot in your schema.  That should really be used to specify which member of a complex is doing the business.  In this case, there is no complex, and every member of the Set performs the action directly.  If there is some difference between the members in terms of what they are doing there, then they shouldn't be in a Set object.

I would suggest that what you should do is just remove the active unit declaration here.  Given the interpretation that a set is basically a shorthand way of saying the same things about each of its components, we would end up saying that each protein in that Set performs the catalysis.  Without the active unit declaration, the Set would be processed as you say: MAPK10 enables JUN kinase activity, MAPK9 enables JUN kinase activity etc.

**Comment by deustp01** [comment]
@goodb That's better - if the set notation itself forces the correct parsing, I'll remove the redundant / inappropriate activeUnit annotations.

**Comment by goodb** [comment]
If it doesn't, then its my problem!  Either way you are done ;).

**Comment by deustp01** [comment]
@goodb - cleanup of events with multiple active units. If I want to assert that either of two members of a protein family M1 and M2 is capable of joining a given complex with some other proteins A, B, and C and functioning as the active unit of a molecular function enabled by the complex, will it work to make a set [M1 or M2], annotate a complex whose components are A and B and C and [M1 or M2], and then annotate the set [M1 or M2] as the active unit?

If that will work, I will use that strategy to patch at least one reaction so far. If that strategy will not work or may come with other risks or difficulties (it looks close to the request you disallowed on Friday), I will simply annotate the complex as mediating the molecular function without specifying an active unit, and leave the issue of capturing the lost bit of information for version 2.

**Comment by goodb** [comment]
@deustp01 I think that will work, please go for it.

**Comment by deustp01** [comment]
> Here is a list of all of the reactions with the multiple enabler problem (based on curator-version BioPAX from Jan 8, 2020):
> [Multiple Enablers.txt](https://github.com/geneontology/pathways2GO/files/4118633/Multiple.Enablers.txt)

I have gone through the list and resolved most of them, mostly by findng a reason to delete all or all but one of the entities listed as an active unit. In a few cases I've had to refer an instance to a curator for action because I was uncomfortable unilaterally resolving it. The Excel file is Ben's list with green color indicating fully resolved (I hope) instances and yellow indicating pending ones. The number in the last column refers to entries in the Word document attached here that give the rationales for what I did in each case.  I will now try to turn those notes into a better organized proposal for how to use the active unit slot in Reactome catalyst activity instances for discussion both among ourselves and with Reactome curators.
[events_with_mutiple_enablers_20200127.xlsx](https://github.com/geneontology/pathways2GO/files/4174835/events_with_mutiple_enablers_20200127.xlsx)
[events_with_mutiple_enablers_20200131.docx](https://github.com/geneontology/pathways2GO/files/4174837/events_with_mutiple_enablers_20200131.docx)

**Comment by goodb** [comment]
@deustp01 it looks like we are pretty close here.  When do you think I will have access to a BioPAX export that contains all of these updates?

**Comment by ukemi** [comment]
@deustp01, when I look at the annotations coming from the reactome gaf in AmiGO2, I only see NSF1 annotated to GO:0031071. NSF1, FXN, ISD1 (LYRM4) and ISCU are annotated to mitochondrial matrix from this pathway. Do you know how NSF1 is chosen from the R-HSA-1362401 complex? Am I looking at the pathway correctly? It seems to me that the complex has the catalyst activity and somehow in the gaf you have identified NSF1 catalytic subunit.

**Comment by deustp01** [comment]
@ukemi The released version of this reaction has two catalystActivities, which I think causes the multiple-enablers issue. It has now been re-annotated to give it one catalyst activity (2Iron:FXN:NFS1:ISD11:ISCU [mitochondrial matrix] complex, with activeUnit PXLP-K258-NFS1-1 [mitochondrial matrix] (NFS1 protein covalently modified with pyridoxal phosphate), mediates GO:0031071 "cysteine desulfurase activity"), positively regulated by the physical entity that was previously annotated as the second catalyst. Bruce May, who curated the original event and the patch says it is a pretty good fit to the biology.

You can see the revised event on our internal site [here](https://curator.reactome.org/PathwayBrowser/#/R-HSA-1362409&SEL=R-HSA-1362408&PATH=R-HSA-1430728). It will propagate into our public database, GAF, and BioPax export naturally with our next release in mid-March and should be available as BioPax sooner (@goodb) depending on Guanming's workload. I will ask about that.

**Comment by ukemi** [comment]
Got it! Thanks @deustp01. I'm still curious as the the origin of the conventional annotation of NFS1 to cysteine desulfurase activity (GO:0031071) and none of the other members of the complex though. This is off topic for this ticket, but I am curious.

**Comment by goodb** [comment]
Since the March release, recent updates to the conversion code and to the shex schema, we are now down to 8 models that don't validate.  All models pass OWL consistency check.  

5 of them fail because they have molecular function nodes with more than one enabling entity: http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-6782135
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-6807505
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-4615885
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-6785470
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-8868766

3 of them fail because they have a Photon as the enabler of a molecular function: 
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-2187335
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-2485179
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-2453902

For the latter, the march release seems to have Photon classified as a chemical entity (resulting in the shex violation as chemical entities can't be enablers).  However, this seems to have been changed to the real entry for Photon in CHEBI https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:30212 which will result in it being typed as a subatomic particle - and likely continue to fail the validation for the same reason.  
 
(note that current (April 18, 2020) noctua-dev has an outdated reacto driving it that is making photons into proteins and thus not showing the shex fail).

**Comment by ukemi** [comment]
I've been in discussion with @deustp01 about the photons. Let's bring it up on the next call, April 29. As for the multiple enablers, we will have to have a closer look a these. Is it because a complex has more than one catalytic subunit?  Considering the number of models we have, this really isn't that bad!

**Comment by deustp01** [comment]
I'll look today - multiple-enablers should be fix-able on the Reactome side. Photons - as far as the Reactome data model is concerned, photons can be inputs and in fact should generally (maybe always) not be enablers. (Even input has a problem with materiality, but there's no need to compound it by treating them as catalysts where GO-CAM expects genome-encoded informational macromolecules.

**Comment by deustp01** [comment]
OK. The three enabling photons have been converted from catalysts to input physical entities.
Events with multiple enablers - for local data model reasons, four of the five here may be hard to fix. One looks like a curation mistake that may be quite hard to track down and fix.
So, @goodb , would it be possible for the GO-CAM building process to check for flawed enablers (the only surviving flaw should be multiple ones, but new curation could create new kinds), and discard the flawed enabler attribute, note its identifier so we can check it on our side, and build a GO-CAM with what's left? If that check + complain approach is easy to implement, that looks like it would be a fairly sturdy fix.

**Comment by goodb** [comment]
@deustp01  I think its possible, but not sure on the overall strategy.  It seems like a deeper modification of the incoming model than we have done so far.  An alternative is simply to report any disagreement with the schema (including this enabler problem and many other potential conflicts that could arise) and the reason, and stop with that.  I kind of like that approach as its generally applicable.  

Following down that road, we might imagine an option on the converter to delete any relations violating the schema.  That would only be important if we had software that cared deeply about the schema and an intention to use these models in that software.  I'm not sure that either of those conditions apply right now.

**Comment by deustp01** [comment]
If I understand right, the number of remaining problem cases is small and most of the ones we know about require fixes on the Reactome side that are hard to do. We can also expect that new content added to Reactome will come with occasional items that violate GO-CAM rules. I was imagining the remaining problems were a major block to using the models in current applications, and if that's not true then the reports you describe are all we need now.

**Comment by ukemi** [comment]
"Hard to do" says to me "Maybe not wrong" and we might want to have a look at our rules. But you are correct. it's a small number of cases.

**Comment by deustp01** [comment]
"Hard to do" is one of those context-dependent terms. Of the five non-photon cases that Ben flagged on April 18 above, one was a curation mistake (the active unit of a catalytic complex was not a component of that complex - now fixed; visible in June). The other four were black box events each of which describe multiple catalytic steps with known catalysts that occur in unknown order, so multiple enablers are crammed into a single event. This tactic preserves all the gene product - molecular function associations but breaks our logic (harmlessly) and GO-CAM's (not so harmlessly). Even if it were possible, it seems unlikely that a rule change on the GO / GO-CAM side is good here. On our side, we can go back to the biology and look for ways to salvage as much information as possible in a rule-compliant way.

The photon cases look like a nice example of "maybe not wrong" - as long as we don't call them (or anything else that's not a gene product) enablers, but use them as inputs, outputs, and the non-gene-product physical entities that mediate regulation instances*, the remaining bit of Reactome - GO disconnect can be ignored safely.
(*If a small molecule like fructose-2,6-bisphosphate can regulate without breaking anything, then perhaps a photon could do it as well.)

**Comment by ukemi** [comment]
As we move forward though, we might want to devise a way to get annotations from those black boxes.

**Comment by deustp01** [comment]
Definitely - part of the larger discussion of aligning data models better for version 2.

**Comment by goodb** [comment]
From the conversion code side, I believe we are not planning to make any more changes.  There will be future changes on the reactome side, but no need for this issue as these will be ongoing with each release I imagine.
